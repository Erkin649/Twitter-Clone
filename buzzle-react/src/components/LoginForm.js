import React from 'react';
import { Redirect } from "react-router";

class LoginForm extends React.Component {
    componentWillUnmount() {
        this.props.unloadLoginForm();
        this.props.clearError();
    }

    handleChange = event => {
        this.props.updateLoginForm(event.target.name, event.target.value);
    }

    handleSubmit = event => {
        event.preventDefault();

        this.props.loginUser({
            'login': this.props.login,
            'password': this.props.password
        });
    }

    render() {
        const user = this.props.user;
        const login = this.props.login;
        const password = this.props.password;

        return (user ?
            <Redirect to="/" />
            :
            <div className="row justify-content-center">
                <div className="col-lg-6 px-5">
                    <form
                        className="p-3"
                        method="POST"
                        action="/login"
                        onSubmit={this.handleSubmit}>
                        <div className="form-group">
                            <label htmlFor="login">Login:</label>
                            <input
                                id="login"
                                className="form-control"
                                name="login"
                                type="text"
                                onChange={this.handleChange}
                                value={login}
                                required />
                        </div>
                        <div className="form-group">
                            <label htmlFor="password">Password:</label>
                            <input
                                id="password"
                                className="form-control"
                                name="password"
                                type="password"
                                onChange={this.handleChange}
                                value={password}
                                required />
                        </div>
                        <input className="btn btn-primary" type="submit" value="Login" />
                    </form>        
                </div>
            </div>
        );
    }
}

export default LoginForm;
