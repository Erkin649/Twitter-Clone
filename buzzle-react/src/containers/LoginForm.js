import { connect } from 'react-redux';

import { clearError } from '../actions/error.js';
import { loginUser } from '../actions/user.js';
import {
    updateLoginForm,
    unloadLoginForm
} from '../actions/forms.js';
import LoginForm from '../components/LoginForm.js';

const mapStateToProps = state => ({
    'user': state.user,
    'login': state.loginForm.login,
    'password': state.loginForm.password
});

const mapDispatchToProps = dispatch => ({
    'clearError': () => dispatch(clearError()),
    'loginUser': credentials => dispatch(loginUser(credentials)),
    'updateLoginForm': (name, value) => dispatch(updateLoginForm(name, value)),
    'unloadLoginForm': () => dispatch(unloadLoginForm())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(LoginForm);
