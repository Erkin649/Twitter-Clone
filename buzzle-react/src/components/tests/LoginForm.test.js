import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import LoginForm from '../LoginForm.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LoginForm />, div);
});

it('renders correctly and equal to snapshot', () => {
    const tree = renderer.create(<LoginForm />)
                         .toJSON();

    expect(tree).toMatchSnapshot();
});