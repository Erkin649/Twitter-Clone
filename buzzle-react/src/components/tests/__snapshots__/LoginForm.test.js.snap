// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and equal to snapshot 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-lg-6 px-5"
  >
    <form
      action="/login"
      className="p-3"
      method="POST"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="login"
        >
          Login:
        </label>
        <input
          className="form-control"
          id="login"
          name="login"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
        />
      </div>
      <input
        className="btn btn-primary"
        type="submit"
        value="Login"
      />
    </form>
  </div>
</div>
`;
